### PR TITLE
E2e/feature/ri 4828 rework tree view

### PR DESCRIPTION
**Pull request by vlad-dargel**

### CODE REVIEW

File: tests/e2e/common-actions/browser-actions.ts [hunk 1]
```diff
@@ -39,17 +40,62 @@ export class BrowserActions {
...
+
+    /**
+     * Check tree view structure
+     * @folders name of folders for tree view build
+     * @delimiter string with delimiter value
+     * @commonKeyFolder flag if not patterned keys will be displayed
+     */
+    async checkTreeViewFoldersStructure(folders: string[][], delimiter: string, commonKeyFolder: boolean): Promise<void> {
+        // Verify that all keys that are not inside of tree view doesn't contain delimiter
+        if (commonKeyFolder) {
```
Comment (mariasergeenko) [inline tests/e2e/common-actions/browser-actions.ts:69]: probably isCommonKeyFolder?

File: tests/e2e/common-actions/browser-actions.ts [hunk 2]
```diff
@@ -39,17 +40,62 @@ export class BrowserActions {
...
+            const notPatternedKeys = Selector('[data-testid^="badge"]').parent('[data-testid^="node-item_"]');
+            const notPatternedKeysNumber = await notPatternedKeys.count;
+            for (let i = 0; i < notPatternedKeysNumber; i++) {
+                await t.expect(notPatternedKeys.nth(i).withText(delimiter).exists).notOk('Not contained delimiter keys');
+            }
+        }
+        // Verify that every level of tree view is clickable
```
Comment (mariasergeenko) [inline tests/e2e/common-actions/browser-actions.ts:76]: can we split the method on 2 or 3 separate methods?

File: tests/e2e/common-actions/browser-actions.ts [hunk 3]
```diff
@@ -39,17 +40,62 @@ export class BrowserActions {
...
+            const innerFoldersNumber = folders[i].length;
+            const array: string[] = [];
+            for (let j = 0; j < innerFoldersNumber; j++) {
+                if (j === 0) {
+                    const folderSelector = `[data-testid="node-item_${folders[i][j]}${delimiter}"]`;
+                    array.push(folderSelector);
+                    await t.click(Selector(folderSelector));
+                }
+                else {
+                    const lastSelector = array[array.length - 1].substring(0, array[array.length - 1].length - 2);
```
Comment (mariasergeenko) [inline tests/e2e/common-actions/browser-actions.ts:88]: could you please clarify it for me?

Comment (vlad-dargel) [reply tests/e2e/common-actions/browser-actions.ts:88]: the purpose is to remove the last two characters from the last element in the array

File: tests/e2e/common-actions/browser-actions.ts (hunk 2 above)
Comment (vlad-dargel) [reply tests/e2e/common-actions/browser-actions.ts:76]: done

File: tests/e2e/common-actions/browser-actions.ts (hunk 1 above)
Comment (vlad-dargel) [reply tests/e2e/common-actions/browser-actions.ts:69]: removed this

File: tests/e2e/common-actions/browser-actions.ts [hunk 4]
```diff
@@ -39,17 +40,78 @@ export class BrowserActions {
...
+     * @param array folders selectors
+     * @param folderName name of folder
+     * @param delimiter string with delimiter value
+     */
+    getFolderSelector(array: string[], folderName: string, delimiter: string): string {
+        if (array.length === 0) {
+            return `[data-testid="node-item_${folderName}${delimiter}"]`;
+        }
+
+        const lastSelector = array[array.length - 1].substring(0, array[array.length - 1].length - 2);
```
Comment (mariasergeenko) [inline tests/e2e/common-actions/browser-actions.ts:85]: but why do you need to remove 2 char?

Comment (vlad-dargel) [reply tests/e2e/common-actions/browser-actions.ts]: removed this method

**Review by mariasergeenko — APPROVED** [review]

**Pull request merged**